src_erl/Communication_Layer/http_Nerlserver/src/Client/clientStatem.erl

```diff
@@ -149,18 +175,27 @@ idle(cast, In = {custom_worker_message, {From, To}}, State = #client_statem_stat
   {keep_state, State};
 
 idle(cast, In = {statistics}, State = #client_statem_state{ myName = MyName, etsRef = EtsRef}) ->
-  sendStatistics(EtsRef),
   ets:update_counter(EtsRef, msgCounter, 1), % last param is increment value
   ets:update_counter(EtsRef, infoIn, nerl_tools:calculate_size(In)),
+  sendStatistics(EtsRef),
   {next_state, idle, State};
 
 idle(cast, In = {training}, State = #client_statem_state{etsRef = EtsRef}) ->
```

Comment:
Avoid such short names for variables. 
More informative names.
InputMessage as an example is much better.